velox/duckdb/conversion/tests/DuckWrapperTest.cpp

```diff
@@ -188,14 +207,47 @@ TEST_F(BaseDuckWrapperTest, tpchSF1) {
   // test TPC-H loading and querying SF0.01
   execute("CALL dbgen(sf=0.01)");
   // test conversion of date, decimal and string
-  verifyUnaryResult<double>("SELECT l_discount FROM lineitem LIMIT 1", {0.04});
+  verifyUnaryResult<ShortDecimal>(
+      "SELECT l_discount FROM lineitem LIMIT 1", {ShortDecimal(4)});
   verifyUnaryResult<Date>(
       "SELECT l_shipdate FROM lineitem LIMIT 1", {Date(9568)});
   verifyUnaryResult<StringView>(
       "SELECT l_comment FROM lineitem LIMIT 1",
       {StringView("egular courts above the")});
 }
 
+TEST_F(BaseDuckWrapperTest, duckToVeloxDecimal) {
+  // test SMALLINT conversion.
+  verifyDuckToVeloxDecimal<ShortDecimal>(
+      "select * from (values ('1.2'::decimal(2,1)),"
+      "('2.2'::decimal(2,1)),('-4.2'::decimal(2,1)))",
+      {ShortDecimal(12), ShortDecimal(22), ShortDecimal(-42)});
+
+  // test INTEGER conversion.
+  verifyDuckToVeloxDecimal<ShortDecimal>(
+      "select * from (values ('1111.1111'::decimal(8,4)),"
+      "('2222.2222'::decimal(8,4)),('-3333.3333'::decimal(8,4)))",
+      {ShortDecimal(11111111),
+       ShortDecimal(22222222),
+       ShortDecimal(-33333333)});
+
+  // test BIGINT conversion.
+  verifyDuckToVeloxDecimal<ShortDecimal>(
+      "select * from (values ('-111111.111111'::decimal(12,6)),"
+      "('222222.222222'::decimal(12,6)),('333333.333333'::decimal(12,6)))",
+      {ShortDecimal(-111111111111),
+       ShortDecimal(222222222222),
+       ShortDecimal(333333333333)});
+
+  verifyDuckToVeloxDecimal<LongDecimal>(
+      "select * from (values ('12345678901234.789'::decimal(18,3) * 10000.555::decimal(20,3)),"
```

Comment:
Please refer this issue for expressing LongDecimals as a product of ShortDecimal literals:
https://github.com/facebookincubator/velox/issues/1771